spec.bs

```diff
@@ -676,3 +676,190 @@ To <dfn>determine the numeric value</dfn> of a [=signal base value=]
+the form of key-value pairs (i.e. contributions to a histogram). The information
+embedded in the contributions is arbitrary but can include things like browsing
+history and other cross-site activity. The API aims to provide some protection
+for this information from being passed from one site to another.
+
+### Restricted contribution processing ### {#restricted-contribution-processing}
+
+The histogram contributions are not exposed directly. Instead, they are
+encrypted so that they can only be processed by a trusted aggregation service.
+This trusted aggregation service sums the values across the reports for each key
+and adds noise to each of these values to produce ‘summary reports’.
+
+The output of that processing will be an aggregate, noised histogram. The
+service ensures that any report can not be processed multiple times. Further,
+information exposure is limited by contribution budgets on the client. In
+principle, this framework can support specifying a noise parameter which
+satisfies differential privacy.
+
+### Unencrypted metadata ### {#unencrypted-metadata}
+
+These reports also expose a limited amount of metadata, which is not based on
+cross-site data. The recipient of the report may also be able to observe
+side-channel information such as the time when the report was sent, or IP
+address of the sender.
+
+### Protecting against leaks via the number of reports ### {#protecting-against-leaks-via-the-number-of-reports}
+
+However, the number of reports with the given metadata could expose some
+cross-site information. To protect against this, the API delays sending reports
+by a randomized amount of time to make it difficult to determine whether a
+report was sent or not from any particular event. In the case that a context ID
+is supplied, the API makes the number of reports sent deterministic (sending
+reports containing no contributions in the payloads if necessary). Additional
+mitigations may also be possible in the future, e.g. adding noise to the report
+count.
+
+### Privacy parameters ### {#privacy-parameters}
+
+The amount of information exposed by this API is a product of the privacy
+parameters used (e.g. contribution limits and the noise distribution used in the
+aggregation service). While we aim to minimize the amount of information
+exposed, we also aim to support a wide range of use cases. The privacy
+parameters are left [=implementation-defined=] to allow different and evolving
+choices in the tradeoffs between information exposure and utility.
+
+Clearing site data {#clearing-site-data}
+----------------------------------------
+
+The [=aggregatable report cache=] as well as any contribution history data
+stored for the [=PrivateAggregation/consume budget if permitted=] algorithm
+contain data about a user’s web activity. As such, the user agent is strongly
+encouraged to expose controls that allow the user to delete data from them.
```

Comment:
I'd prefer a MUST language in the normative parts of the spec. (similar to https://wicg.github.io/client-hints-infrastructure/#accept-ch-cache-definition)

Reply:
The privacy considerations section was marked as non-normative, but I've switched this part to be normative (and use "must"). Not sure if it makes sense to switch the key rotation recommendation (under "Protecting the histogram contributions") to a normative "should" as well.

Reply:
I think it'd be better to move these MUSTs and SHOULDs to the normative parts that define the cache, and then have the P&S section non-normative and just clarify that these concerns are handled by those normative sections.

Reply:
Makes sense -- moved the normative clearing data section under 'Storage'